tests/viadot_tests/integration/test_eurostat.py

```diff
@@ -0,0 +1,226 @@
+import logging
+
+import pandas as pd
+import pytest
+from viadot.sources import Eurostat
+
+
+def test_and_validate_dataset_code_without_params(caplog):
+    """This function is designed to test the accuracy of the data retrieval feature in a program.
+    Specifically, it tests to ensure that the program returns a non-empty DataFrame when a correct
+    dataset code is provided without any parameters. The function is intended to be used in software
+    development to verify that the program is correctly retrieving data from the appropriate dataset.
+    """
+    task = Eurostat().to_df(dataset_code="ILC_DI04")
+
+    assert isinstance(task, pd.DataFrame)
+    assert not task.empty
+    assert caplog.text == ""
+
+
+def test_wrong_dataset_code_logger(caplog):
+    """This function is designed to test the accuracy of the error logging feature in a program.
+    Specifically, it tests to ensure that the program is able to correctly identify and log errors
+    when provided with only incorrect dataset code.
+    The function is intended to be used in software development to identify correct type errors
+    and messages in the program's handling of codes.
+    """
+    task = Eurostat()
+
+    with pytest.raises(ValueError, match="DataFrame is empty!"):
+        with caplog.at_level(logging.ERROR):
+            task.to_df(dataset_code="ILC_DI04E")
+    assert (
+        "Failed to fetch data for ILC_DI04E, please check correctness of dataset code!"
+        in caplog.text
+    )
```

Comment:
In general it's cleaner to validate exceptions rather than logging output, I would treat validating logs as a last resort (it's used to validate that the logging is correct, not that the code's logic is correct).

Reply:
I am validating exceptions and logging at the same time